language/bytecode-verifier/src/signature.rs

```diff
@@ -7,284 +7,212 @@
 use libra_types::vm_error::{StatusCode, VMStatus};
 use vm::{
     access::ModuleAccess,
-    errors::append_err_info,
-    file_format::{Bytecode, CompiledModule, Kind, SignatureToken, StructFieldInformation},
+    errors::{append_err_info, VMResult},
+    file_format::{
+        Bytecode, CompiledModule, Kind, SignatureIndex, SignatureToken, StructFieldInformation,
+        TableIndex,
+    },
     IndexKind,
 };
 
 pub struct SignatureChecker<'a> {
     module: &'a CompiledModule,
+    signature_context: Option<Vec<Kind>>,
```

Comment:
not a big fan of this field story... wish we did not change it this much for this refactor PR and I was planning on reviewing this anyway, but I guess it's all good... :)

Reply:
Yeah sorry, I started by trying not to do that, but the massive amount of chaining/fusing/collecting made my brain hurt. And there was a ton of duplicated code all over. It was honestly easier to do this